packages/amplify_core/lib/src/types/storage/copy_options.dart

```diff
@@ -12,11 +12,17 @@ class StorageCopyOptions
         AWSSerializable<Map<String, Object?>>,
         AWSDebuggable {
   /// {@macro amplify_core.storage.copy_options}
-  const StorageCopyOptions({this.pluginOptions});
+  const StorageCopyOptions({
+    this.pluginOptions,
+    this.buckets,
+  });
 
   /// plugin specific options for `Amplify.Storage.copy`.
   final StorageCopyPluginOptions? pluginOptions;
 
+  /// Optionally specify which bucket to target
+  final CopyBuckets? buckets;
```

Comment:
Q: Does JS also use `CopyBuckets buckets` for copy API optinal param?

Reply:
No they have a different [copy signature](https://docs.amplify.aws/nextjs/build-a-backend/storage/copy-files/#specify-a-bucket-or-copy-across-buckets--regions) that is more extensible:

- source
  - path
  - bucket
- destination
   - path
   - bucket

I prefer their signature over ours, but it would be a breaking change to wrap our source/destination parameters in an object, so I think we stick with the CopyBuckets approach.